modal/app.py

```diff
@@ -723,6 +723,13 @@ def wrapper(user_cls: CLS_T) -> _Cls:
             ):
                 raise InvalidError("A class must have `enable_memory_snapshot=True` to use `snap=True` on its methods.")
 
+            # Disallow enable_memory_snapshot for parameterized classes
+            constructor = dict(inspect.getmembers(user_cls, inspect.isfunction)).get("__init__")
+            if enable_memory_snapshot and constructor:
+                params = inspect.signature(constructor).parameters
+                if len(params) > 1:
+                    raise InvalidError("Cannot use parameterized classes with `enable_memory_snapshot=True`.")
```

Comment:
I think it'll be more user friendly if the error message includes the name of the class so the user can more quickly go to the problematic source.

Reply:
I think this code block should also have a TODO which references a Linear ticket (MOD-123) about supporting this properly.